File: app/src/main/java/com/sileneer/hydrogenbrowser/TitleLayout.java
package com.sileneer.hydrogenbrowser;

import android.app.Activity;
import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.constraintlayout.widget.ConstraintLayout;

public class TitleLayout extends ConstraintLayout {

    private static TextView titleText;

    public TitleLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
        LayoutInflater.from(context).inflate(R.layout.title, this);
        ImageView titleBack = findViewById(R.id.title_back);
        titleBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ((Activity) getContext()).finish();
            }
        });
        titleText = findViewById(R.id.title_text);
    }

    protected static void setTitleText(String str){
        titleText.setText(str);
    }
}

File: app/src/main/java/com/sileneer/hydrogenbrowser/common/utils/TitleLayout.kt
package com.sileneer.hydrogenbrowser.common.utils

import android.app.Activity
import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import androidx.constraintlayout.widget.ConstraintLayout
import com.sileneer.hydrogenbrowser.databinding.TitleBinding

class TitleLayout(context: Context, attrs: AttributeSet) : ConstraintLayout(context, attrs) {

    private val binding: TitleBinding

    init {
        binding = TitleBinding.inflate(LayoutInflater.from(context), this, true)
        binding.titleBack.setOnClickListener {
            (getContext() as? Activity)?.finish()
        }
    }

    fun setTitleText(str: String) {
        binding.titleText.text = str
    }
}
